errbot/backends/irc.py

```diff
@@ -186,6 +192,7 @@ def leave(self, reason=None):
             reason = ""
 
         self.connection.part(self.room, reason)
+        del self._bot._rooms[self.room]  # Destroy the room that we're leaving
```

Comment:
We're deleting an entry from the rooms dict so it's obvious what this code is doing. Comments should generally not describe *what* code is doing (as that should be obvious), only *why* it is doing it if this isn't immediately clear.

Also, race condition on `self._bot._rooms` again. :smile: